deltaconcor/src/main/java/uk/gov/justice/laadces/deltaconcor/Runner.java

```diff
@@ -0,0 +1,353 @@
+        return visitForwards(startId, endId, concorContribution -> {
+            final long maatId = concorContribution.repId();
+            final CONTRIBUTIONS dto = xmlTransformService.fromXML(concorContribution.fullXml());
+            if (dto == null) {
+                log.warn("generateDailyCounts: Failed to transform XML for concorId {}, maatId {}", concorContribution.id(), maatId);
+            } else {
+                final int dayIndex = (int) ChronoUnit.DAYS.between(startDate, concorContribution.dateCreated());
+                if (dayIndex < 0 || dayIndex >= newChanges.size()) {
+                    log.warn("generateDailyCounts: {} out of range for concorId {}, maatId {}", concorContribution.dateCreated(), concorContribution.id(), maatId);
+                    return true;
+                }
+                if (dto.getFlag().equalsIgnoreCase("new")) {
+                    final ChangeLog log = new ChangeLog(concorContribution.dateCreated().toString());
+                    log.setMaatId(maatId);
+                    log.setNext_concorContributionId(concorContribution.id());
+                    log.setSentRecords(1);
+                    if (changeService.compare(null, dto, log)) {
+                        log.setChangedRecords(1);
+                    }
+                    changeLogs.add(log);
+                    changeService.addTo(newChanges.get(dayIndex), log);
+                } else if (dto.getFlag().equalsIgnoreCase("update")) {
+                    final var prev = maatIdToContribution.get(maatId);
+                    final ChangeLog log = new ChangeLog(concorContribution.dateCreated().toString());
+                    log.setMaatId(maatId);
+                    log.setNext_concorContributionId(concorContribution.id());
+                    log.setPrev_concorContributionId(prev != null ? prev.getId() : -1L);
+                    log.setSentRecords(1);
+                    if (changeService.compare(prev, dto, log)) {
+                        log.setChangedRecords(1);
+                    }
+                    changeLogs.add(log);
+                    changeService.addTo(updChanges.get(dayIndex), log);
+                } else {
+                    log.warn("generateDailyCounts: Unknown flag {} for concorId {}, maatId {}", dto.getFlag(), concorContribution.id(), maatId);
+                }
+                maatIdToContribution.put(maatId, dto);
+            }
+            return true;
+        });
+    }
+
+    /**
+     * Uses an exhaustive query to visit forwards through the ConcorContribution table.
+     *
+     * @param startId start ID to walk forward from
+     * @param endId end ID to finish walking forward until
+     * @param visitor the RecordVisitor to call for each record visited
+     * @return the ID of the last record visited
+     */
+    private long visitForwards(final long startId, final long endId, final RecordVisitor visitor) {
```

Comment:
This return value doesn't seem to be used anywhere, can this be turned into a void method, to make it clearer to understand?

Reply:
yes, it could. I was using the long in one version of the code (not committed), but it's no longer needed.